internal/server/server.go

```diff
@@ -17,209 +17,255 @@ import (
+	slog.InfoContext(ctx, fmt.Sprintf("the MCP server %s has %d registered tools", serveOpts.Name, len(tools)))
 
-		waitForShutdownStreamable(ctx, streamableServer)
-	default:
-		return fmt.Errorf("invalid transport type: %s", serveOpts.Transport)
+	// Start the MCP server depending on the chosen transport.
+	err = handleServerRun(ctx, srv, serveOpts)
+	if err != nil {
+		return err
 	}
 
-	// So, if we reach here, it means graceful shutdown (or forced exit within waitForShutdown).
-	return nil
+	return err
+}
+
+// createMCPServer creates the MCP server, but does not start serving it yet.
+func createMCPServer(ctx context.Context, serveOpts *ServeOptions) *mcpserver.MCPServer {
+	// Create MCP server options.
+	// For additional ones, refer to https://github.com/mark3labs/mcp-go/blob/main/server/server.go
+	opts := []mcpserver.ServerOption{
+		mcpserver.WithLogging(),              // enables logging capabilities for the server
+		mcpserver.WithRecovery(),             // recovers from panics in tool handlers
+		mcpserver.WithToolCapabilities(true), // configures tool-related server capabilities
+	}
+
+	slog.DebugContext(ctx, "the MCP server options have been created",
+		"server.options", fmt.Sprintf("%+v", opts),
+	)
+
+	// Create the MCP server with above options.
+	srv := mcpserver.NewMCPServer(serveOpts.Name, serveOpts.Version, opts...)
+
+	return srv
 }
 
-func createMCPServer(ctx context.Context, serveOpts *ServeOptions) (*mcpserver.MCPServer, error) {
+// handleToolsRegistration loads the OAS file, transforms it into MCP tools and registers them into the MCP server.
+func handleToolsRegistration(
+	ctx context.Context,
+	srv *mcpserver.MCPServer,
+	serveOpts *ServeOptions,
+) (*mcpserver.MCPServer, []string, error) {
 	// Load OpenAPI spec.
 	oasDoc, err := openapi2mcp.LoadOpenAPISpec(serveOpts.OASPath)
 	if err != nil {
-		slog.ErrorContext(ctx, "failed to read the API spec", "error", err)
+		slog.ErrorContext(ctx, "failed to read the API spec",
+			"error", err,
+		)
 
-		return nil, fmt.Errorf("failed to read the API spec: %w", err)
+		return nil, []string{}, fmt.Errorf("failed to read the API spec: %w", err)
 	}
 
-	// Overwrite the Trento URL in the OpenAPI
+	// Overwrite or the Trento URL in the OpenAPI
 	if len(oasDoc.Servers) > 0 {
 		oasDoc.Servers[0].URL = serveOpts.TrentoURL
 	} else {
+		// Or just add it
 		oasDoc.Servers = append(oasDoc.Servers, &openapi3.Server{
 			URL: serveOpts.TrentoURL,
 		})
 	}
 
-	// Create MCP server options.
-	opts := []mcpserver.ServerOption{
-		mcpserver.WithLogging(),
-		mcpserver.WithRecovery(),
-	}
-
-	// Create MCP server.
-	srv := mcpserver.NewMCPServer("trento-mcp-server", oasDoc.Info.Version, opts...)
-
 	// Extract the API operations.
 	operations := openapi2mcp.ExtractOpenAPIOperations(oasDoc)
 
 	// Register them as MCP tools.
-	openapi2mcp.RegisterOpenAPITools(srv, operations, oasDoc, nil)
+	tools := openapi2mcp.RegisterOpenAPITools(srv, operations, oasDoc, nil)
```

Comment:
At this point, we could enrich or modify the tools as we wish. Or, if we happen to change the MCP server dependency, we will only have to change this `handleToolsRegistration` thing.